docs/design/ovs-pipeline.md

```diff
@@ -307,6 +334,28 @@ The table-miss flow entry (flow 3) will drop all other packets. This flow should
 never be used because only ARP traffic should go to this table, and
 ARP traffic will either match flow 1 or flow 2.
 
+### ServiceHairpinTable (23)
+
+If a Pod accesses a Service and the Pod is selected as the Endpoint of the Service,
+the source IP should be SNAT'd with a virtual hairpin IP in table [hairpinSNATTable].
+The source and destination IP addresses cannot be the same, otherwise the connection
+will be broken. The reason will be explained in detail in chapter table [hairpinSNATTable].
+For response packets, the destination IP is the virtual hairpin IP, so the destination IP
+should be changed back to the IP of the Pod so that the packets can be forwarded back
+correctly.
+
+If you dump the flows for this table, you should see the flows:
+
+```text
+1. table=23, priority=200,ip,nw_dst=169.254.169.252 actions=move:NXM_OF_IP_SRC[]->NXM_OF_IP_DST[],load:0x1->NXM_NX_REG0[18],resubmit(,30)
+2. table=23, priority=0 actions=resubmit(,24)
+```
+
+Flow 1 is used to match the packet whose destination IP is the virtual hairpin IP and
+change the destination IP by loading register `NXM_OF_IP_SRC` to `NXM_OF_IP_DST`.
+Bit 18 in NXM_NX_REG0 is set to 0x1, which indicates the packet should be output
+on the port on which it was received, and it is consumed by table [L2ForwardingOutTable].
```

Comment:
on the port -> to the port

Reply:
and it is consumed by -> which is done in table

Reply:
Thanks, updated